lib/ansible/plugins/connection/psrp.py

```diff
@@ -0,0 +1,600 @@
+# Copyright (c) 2018 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = """
+author: Ansible Core Team
+connection: psrp
+short_description: Run tasks over Microsoft PowerShell Remoting Protocol
+description:
+- Run commands or put/fetch on a target via PSRP (WinRM plugin)
+- This is similar to the I(winrm) connection plugin which uses the same
+  underlying transport but instead runs in a PowerShell interpreter.
+version_added: "2.7"
+requirements:
+- pypsrp (Python library)
+options:
+  # transport options
+  remote_addr:
+    description:
+    - The hostname or IP address of the remote host.
+    default: inventory_hostname
+    vars:
+    - name: ansible_host
+    - name: ansible_psrp_host
+  remote_user:
+    description:
+    - The user to log in as.
+    vars:
+    - name: ansible_user
+    - name: ansible_psrp_user
+  port:
+    description:
+    - The port for PSRP to connect on the remote target.
+    - Default is C(5986) if I(protocol) is not defined or is C(https),
+      otherwise the port is C(5985).
+    vars:
+    - name: ansible_port
+    - name: ansible_psrp_port
+  protocol:
+    description:
+    - Set the protocol to use for the connection.
+    - Default is C(https) if I(port) is not defined or I(port) is not C(5985).
+    choices:
+    - http
+    - https
+    vars:
+    - name: ansible_psrp_protocol
+  path:
+    description:
+    - The URI path to connect to.
+    vars:
+    - name: ansible_psrp_path
+    default: 'wsman'
+  auth:
+    description:
+    - The authentication protocol to use when authenticating the remote user.
+    - The default, C(negotiate), will attempt to use C(Kerberos) if it is
+      available and fall back to C(NTLM) if it isn't.
+    vars:
+    - name: ansible_psrp_auth
+    choices:
+    - basic
+    - certificate
+    - negotiate
+    - kerberos
+    - ntlm
+    - credssp
+    default: negotiate
+  cert_validation:
```

Comment:
Isn't the option for modules `validate_certs` ? It feels strange to introduce something new here.

Looking through sources I can only see *module_utils/azure_rm_common.py* doing this.

Reply:
For the **httpapi** plugin the parameter is `validate_certs` and config item `ansible_httpapi_validate_certs`.

Reply:
This was mostly to replicate the actual pypsrp library but I don't mind changing it. In reality people would be setting `ansible_psrp_cert_validation` and not just `cert_validation` so not the exactly the same as module options. Probably makes sense to have this as a similar name though.

Reply:
TBH I prefer `cert_validation` as it uses namespace prefixes (and related options sort together). Maybe we ought to fix the **httpapi** connection plugin instead ? ;-)